src/kernel/arch/x86/syscalls.zig

```diff
@@ -257,78 +359,115 @@ pub fn init() void {
+fn testHandler4(ctx: *arch.CpuState, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize) syscalls.Error!usize {
     test_int += arg1 + arg2 + arg3 + arg4;
     return 4;
 }
 
-fn testHandler5(ctx: *arch.CpuState, arg1: u32, arg2: u32, arg3: u32, arg4: u32, arg5: u32) u32 {
+fn testHandler5(ctx: *arch.CpuState, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize) syscalls.Error!usize {
     test_int += arg1 + arg2 + arg3 + arg4 + arg5;
     return 5;
 }
 
+fn testHandler6(ctx: *arch.CpuState, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize) syscalls.Error!usize {
+    return syscalls.Error.OutOfMemory;
+}
+
 test "registerSyscall returns SyscallExists" {
-    registerSyscall(123, testHandler0) catch unreachable;
-    registerSyscall(123, testHandler0) catch |err| {
+    registerSyscall(122, testHandler0) catch unreachable;
+    registerSyscall(122, testHandler0) catch |err| {
         return;
     };
     expect(false);
 }
 
 fn runtimeTests() void {
-    registerSyscall(123, testHandler0) catch panic(@errorReturnTrace(), "FAILURE registering handler 0\n", .{});
-    registerSyscall(124, testHandler1) catch panic(@errorReturnTrace(), "FAILURE registering handler 1\n", .{});
-    registerSyscall(125, testHandler2) catch panic(@errorReturnTrace(), "FAILURE registering handler 2\n", .{});
-    registerSyscall(126, testHandler3) catch panic(@errorReturnTrace(), "FAILURE registering handler 3\n", .{});
-    registerSyscall(127, testHandler4) catch panic(@errorReturnTrace(), "FAILURE registering handler 4\n", .{});
-    registerSyscall(128, testHandler5) catch panic(@errorReturnTrace(), "FAILURE registering handler 5\n", .{});
+    registerSyscall(121, testHandler6) catch panic(@errorReturnTrace(), "FAILURE registering handler 6\n", .{});
+    registerSyscall(122, testHandler0) catch panic(@errorReturnTrace(), "FAILURE registering handler 0\n", .{});
+    registerSyscall(123, testHandler1) catch panic(@errorReturnTrace(), "FAILURE registering handler 1\n", .{});
+    registerSyscall(124, testHandler2) catch panic(@errorReturnTrace(), "FAILURE registering handler 2\n", .{});
+    registerSyscall(125, testHandler3) catch panic(@errorReturnTrace(), "FAILURE registering handler 3\n", .{});
+    registerSyscall(126, testHandler4) catch panic(@errorReturnTrace(), "FAILURE registering handler 4\n", .{});
+    registerSyscall(127, testHandler5) catch panic(@errorReturnTrace(), "FAILURE registering handler 5\n", .{});
 
     if (test_int != 0) {
         panic(@errorReturnTrace(), "FAILURE initial test_int not 0: {}\n", .{test_int});
     }
 
-    if (syscall0(123) != 0 or test_int != 1) {
-        panic(@errorReturnTrace(), "FAILURE syscall0\n", .{});
+    if (syscall0(122)) |res| {
+        if (res != 0 or test_int != 1) {
+            panic(@errorReturnTrace(), "FAILURE syscall0\n", .{});
+        }
+    } else |e| {
+        panic(@errorReturnTrace(), "FAILURE syscall0 errored\n", .{});
```

Comment:
Could also add the error to the print. Same for the others

Reply:
Good idea :+1: